test/commands/deploy/metadata/resume.nut.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as path from 'path';
+import { SourceTestkit } from '@salesforce/source-testkit';
+import { DeployResultJson } from '../../../../src/utils/types';
+
+describe('deploy metadata resume NUTs', () => {
+  let testkit: SourceTestkit;
+
+  before(async () => {
+    testkit = await SourceTestkit.create({
+      repository: 'https://github.com/salesforcecli/sample-project-multiple-packages.git',
+      executable: path.join(process.cwd(), 'bin', 'dev'),
+      nut: __filename,
+    });
+  });
+
+  after(async () => {
+    await testkit?.clean();
+  });
+
+  describe('--use-most-recent', () => {
+    it('should resume most recently started deployment', async () => {
+      await testkit.execute<DeployResultJson>('deploy:metadata', {
+        args: '--source-dir force-app --async',
+        json: true,
+        exitCode: 0,
+      });
```

Comment:
suggestion: assert something about the state of the cache between async/resume and after resume

I just used fs.readFile